torch/testing/asserts.py

```diff
@@ -0,0 +1,186 @@
+import functools
+import inspect
+import warnings
+from typing import Callable, Optional, Tuple
+
+import torch
+from torch.testing._internal.common_utils import get_comparison_dtype as _get_comparison_dtype, TestCase as _TestCase
+
+__all__ = ["assert_tensors_equal", "assert_tensors_allclose"]
+
+try:
+    import pytest
+
+    UsageError = pytest.UsageError
+except ImportError:
+
+    class UsageError(Exception):
+        pass
+
+
+class UsageWarning(Warning):
+    pass
+
+
+def _get_default_rtol_and_atol(tensor1: torch.Tensor, tensor2: torch.Tensor) -> Tuple[float, float]:
+    return _TestCase()._getDefaultRtolAndAtol(tensor1.dtype, tensor2.dtype)
+
+
+def _hide_internal_traceback(fn):
+    @functools.wraps(fn)
+    def wrapper(*args, **kwargs):
+        try:
+            return fn(*args, **kwargs)
+        except AssertionError:
+            for frame_info in inspect.trace():
+                frame_info.frame.f_locals.setdefault("__tracebackhide__", True)
+            raise
+
+    return wrapper
+
+
+# These attributes will be checked by default in every call to assert_tensors_(equal|allclose) and
+# assert_(equal|allclose) if the inputs are tensors.
+DEFAULT_CHECKED_ATTRIBUTES = (
+    "shape",
+    "dtype",
+)
```

Comment:
We should probably move `shape` out of this list to make it non-optional. Otherwise we end up in cases where we want to check the values of two tensors without matching shapes.

Reply:
Agreed. It's hard to imagine a scenario where we want to support mismatched shapes. 

Other default candidates: device, strides

Reply:
As I think about this, I suggest we add device and strides.

Reply:
We aren't currently checking the `device` by default

https://github.com/pytorch/pytorch/blob/1772e26f6380d189ba4e03da492ff18eaa139563/torch/testing/_internal/common_utils.py#L1092-L1094

Reply:
Yes but the TODO right above that line says we should:

https://github.com/pytorch/pytorch/blob/1772e26f6380d189ba4e03da492ff18eaa139563/torch/testing/_internal/common_utils.py#L1091